pkg/kubelet/pod_workers_test.go

```diff
@@ -303,15 +303,15 @@ func TestUpdatePodWithTerminatedPod(t *testing.T) {
 		UpdateType: kubetypes.SyncPodKill,
 		RunningPod: runningPod,
 	})
-
+	drainAllWorkers(podWorkers)
 	if podWorkers.IsPodKnownTerminated(pod.UID) == true {
 		t.Errorf("podWorker state should not be terminated")
 	}
 	if podWorkers.IsPodKnownTerminated(terminatedPod.UID) == false {
 		t.Errorf("podWorker state should be terminated")
 	}
-	if podWorkers.IsPodKnownTerminated(runningPod.ID) == true {
-		t.Errorf("podWorker state should not be marked terminated for a running pod")
+	if podWorkers.IsPodKnownTerminated(runningPod.ID) == false {
```

Comment:
Yeah, the `RunningPod` field is only populated by the kubelet when cleaning up orphaned containers:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_pods.go#L1118-L1134

I'm not sure what the original intent was for this test, but terminating the orphaned pod represented by this is expected. It looks like the running pod case is already covered by the variable `pod`, so I think maybe this case can just be deleted?

Reply:
runningPod and pod are different type, maybe the original intent was test podWorkers works for both type of pod. But the other promblem here is the original test didn't call 	drainAllWorkers(podWorkers) befor assert pod status.

Reply:
Right, I guess if we want to leave the test case in place, then it's expected that it's terminated, which is what you have here. It might help clear things up to rename `runningPod` to `orphanedPod` (suggestion below).